frontend/src/components/header.tsx

```diff
@@ -1,13 +1,19 @@
 import { useNavigate } from "@tanstack/react-router";
 import "../styles/header.css";
 import Navbar from "./navbar.tsx";
+import { useState } from "react";
+import HamburgerButton from "./hamburger/hamburger-button.tsx";
+import HamburgerMenu from "./hamburger/hamburger-sidemenu.tsx";
 
 const Header = () => {
   const navigate = useNavigate();
   const handleLogoClicked = () => {
     navigate({ to: "/" });
   };
 
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  
+
```

Comment:
It would be cleaner if you had `handleClose` and `handleOpen`

```suggestion
  const handleClick = setIsMenuOpen((prev) => !prev)
```